mapie/utils.py

```diff
@@ -654,3 +666,369 @@ def compute_quantiles(vector: NDArray, alpha: NDArray) -> NDArray:
+        The "true" values, target for the calibrator.
+    y_score : NDArray of shape (n_samples,)
+        The scores given from the calibrator.
+    num_bins : int
+        Number of bins to make the split in the y_score.
+    strategy : str
+        The way of splitting the predictions into different bins.
+    Returns
+    -------
+    Union[NDArray, NDArray, NDArray, NDArray]
+    - [0]: NDArray of shape (num_bins,)
+    An array of all the splitting points for a new bin.
+    - [1]: NDArray of shape (num_bins,)
+    An array of the average accuracy in each of the bins.
+    - [2]: NDArray of shape (num_bins,)
+    An array of the average confidence in each of the bins.
+    - [3]: NDArray of shape (num_bins,)
+    An array of the number of observations in each of the bins.
+    """
+    bins = get_binning_groups(y_score, num_bins, strategy)
+    binned = np.digitize(y_score, bins, right=True)
+    bin_accs = np.zeros(num_bins)
+    bin_confs = np.zeros(num_bins)
+    bin_sizes = np.zeros(num_bins)
+
+    for bin in range(num_bins):
+        bin_sizes[bin] = len(y_score[binned == bin])
+        if bin_sizes[bin] > 0:
+            bin_accs[bin] = np.divide(
+                np.sum(y_true[binned == bin]),
+                bin_sizes[bin],
+            )
+            bin_confs[bin] = np.divide(
+                np.sum(y_score[binned == bin]),
+                bin_sizes[bin],
+            )
+    return bins, bin_accs, bin_confs, bin_sizes  # type: ignore
+
+
+def check_split_strategy(
+    strategy: Optional[str]
+) -> str:
+    """
+    Checks that the split strategy provided is valid
+    and defults None split strategy to "uniform".
+    Parameters
+    ----------
+    strategy : Optional[str]
+        Can be a string or None.
+
+    Returns
+    -------
+    str
+        The spitting strategy that will be adopted, needs to be a string.
+
+    Raises
+    ------
+    ValueError
+        If the strategy is not part of the valid strategies.
+    """
+    if strategy is None:
+        strategy = "uniform"
+    if strategy not in SPLIT_STRATEGIES:
+        raise ValueError(
+            "Please provide a valid splitting strategy."
+        )
+    return strategy
+
+
+def check_number_bins(
+    num_bins: int
+) -> None:
+    """
+    Checks that the bin specified is a number.
+
+    Parameters
+    ----------
+    num_bins : int
+        An integer that determines the number of bins to create
+        on an array.
+
+    Raises
+    ------
+    ValueError
+        When num_bins is not an integer is raises an error.
+
+    ValueError
+        When num_bins is a negative number is raises an error.
+    """
+    if isinstance(num_bins, int) is False:
+        raise ValueError(
+            "Please provide a bin number as an integer."
+        )
+    elif num_bins < 1:
+        raise ValueError(
+            """
+            Please provide a bin number greater than
+            or equal to  1.
+            """
+        )
```

Comment:
if, else

Reply:
I don't believe this is a case where we can use if, else. I have changed it to a double if.

Reply:
I think it would be better to have :
if --> raise Value Error
elif --> Raise Value Error
else --> return numbins